### PR TITLE
Fwk edition integration

### PR DESCRIPTION
**Pull request by sjovdnbos**
https://github.com/openvstorage/openvstorage-health-check/issues/353

### CODE REVIEW

File: ovs/lib/migration.py [hunk 1]
```diff
@@ -390,4 +390,20 @@ def _update_manifest_cache_size(_proxy_config_key):
             except Exception:
                 MigrationController._logger.exception('Integration of stats monkey failed')
 
+
+        ######################################
+        # Introduction of version key
```
Comment (JeffreyDevloo) [inline ovs/lib/migration.py]: Change to edition key

File: ovs/lib/migration.py [hunk 2]
```diff
@@ -390,4 +390,20 @@ def _update_manifest_cache_size(_proxy_config_key):
...
+        if Configuration.get('key', default=False) not in ['community', 'enterprise']:
```
Comment (JeffreyDevloo) [inline ovs/lib/migration.py]: Should be Configuration.get('/ovs/framework/edition', default=False) here. The entry 'key' will never exist (as it is never set)

File: ovs/lib/nodeinstallation.py [hunk 3]
```diff
@@ -972,6 +972,12 @@ def _finalize_setup(client, node_name, node_type):
             storagerouter.rdma_capable = False
         storagerouter.node_type = node_type
         storagerouter.save()
+        if not Configuration.exists('/ovs/framework/edition'):
+            try:
+                val = storagerouter.features['alba']['edition']
+                Configuration.set(key='/ovs/framework/edition', value=val)
+            except TypeError:
```
Comment (JeffreyDevloo) [inline ovs/lib/nodeinstallation.py]: Type error seems weird here.
`storagerouter.features['alba']['edition']` could give off keys error
`Configuration.set` could throw NotFoundException, RuntimeError,...
better to use the default exception so except Exception: ...

**Review by JeffreyDevloo — CHANGES_REQUESTED** [review]

File: ovs/extensions/packages/packagefactory.py [hunk 4]
```diff
@@ -17,22 +17,40 @@
...
+from ovs.extensions.generic.configuration import Configuration
 from ovs_extensions.packages.packagefactory import PackageFactory as _PackageFactory
 
 
 class PackageFactory(_PackageFactory):
     """
     Factory class returning specialized classes
     """
+    universal_packages = ['arakoon', 'openvstorage', 'openvstorage-backend', 'openvstorage-sdm']
+    ose_only_packages = ['alba','volumedriver-no-dedup-base', 'volumedriver-no-dedup-server']
```
Comment (kvanhijf) [inline ovs/extensions/packages/packagefactory.py]: Missing spacing (and below)

File: ovs/extensions/packages/packagefactory.py [hunk 5]
```diff
@@ -17,22 +17,40 @@
...
+            elif edition == 'enterprise':
+                package_names = cls.ee_only_packages
+                binaries = cls.ee_only_binaries
+            else:
+                raise ValueError('Edition could not be found in configuration')
+        else:
+            package_names = cls.ose_only_packages + cls.ee_only_packages
+            binaries = cls.ee_only_binaries + cls.ee_only_binaries
+
+        return {'names': package_names+cls.universal_packages,
```
Comment (kvanhijf) [inline ovs/extensions/packages/packagefactory.py]: Spacing

File: ovs/extensions/packages/packagefactory.py [hunk 6]
```diff
@@ -17,22 +17,40 @@
...
+                package_names = cls.ose_only_packages
+                binaries = cls.ose_only_binaries
+            elif edition == 'enterprise':
+                package_names = cls.ee_only_packages
+                binaries = cls.ee_only_binaries
+            else:
+                raise ValueError('Edition could not be found in configuration')
+        else:
+            package_names = cls.ose_only_packages + cls.ee_only_packages
+            binaries = cls.ee_only_binaries + cls.ee_only_binaries
```
Comment (kvanhijf) [inline ovs/extensions/packages/packagefactory.py]: Twice the EE binaries

File: ovs/lib/migration.py [hunk 7]
```diff
@@ -390,4 +390,20 @@ def _update_manifest_cache_size(_proxy_config_key):
...
+            try:
+                storagerouters = StorageRouterList.get_storagerouters()
+                for sr in storagerouters:
+                    try:
+                        val = sr.features['alba']['edition']
+                        Configuration.set('/ovs/framework/edition', val)
+                        break
+                    except:
+                        MigrationController._logger.exception('StorageRouter {0} did not yield edition value'.format(sr.name))
+                        continue
```
Comment (kvanhijf) [inline ovs/lib/migration.py]: Obsolete `continue`, end of loop anyway

**Review by kvanhijf — CHANGES_REQUESTED** [review]

**Review by JeffreyDevloo — APPROVED** [review]

**Review by kvanhijf — APPROVED** [review]

**Pull request merged**